PythonVisualizations/Array.py

```diff
@@ -258,9 +258,54 @@ def remove(self, val):
             return True
         return False
 
+    def traverse(self):
+        self.cleanUp()
+
+        # draw an index pointing to the first cell
+        indexDisplay = self.createIndex(0, 'j')
+        self.cleanup |= set(indexDisplay)
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        outputBox = self.canvas.create_rectangle(\
+            ARRAY_X0+ CELL_SIZE//2, ARRAY_Y0 + canvasDimensions[1] // 3, ARRAY_X0 + CELL_SIZE*len(self.list), \
+            ARRAY_Y0 + canvasDimensions[1] // 3 + 50,\
+            fill = OPERATIONS_BG)
+        self.cleanup.add(outputBox)
+
+        for j in range(len(self.list)):
+            # calculate where the value will need to move to
+            outputBoxCoords = self.canvas.coords(outputBox)
+            midOutputBox = (outputBoxCoords[3]-outputBoxCoords[1])//2+outputBoxCoords[1]
+
+            # create the value to move to output box
+            valueOutput = self.copyCanvasItem(self.list[j].display_val)
+            valueList = (valueOutput,)
+            self.cleanup.add(valueOutput)
+
+            # move value to output box
+            toPositions = (outputBoxCoords[0] + (j+1)*(CELL_SIZE*3)//4, midOutputBox)
+            self.moveItemsTo(valueList, (toPositions,), sleepTime=.05)
+
+            # make the value 25% smaller
+            newSize = (VALUE_FONT[0], int(VALUE_FONT[1]*.75))
+            self.canvas.itemconfig(valueOutput, font=newSize)
+            self.window.update()
+
+            # wait and then move the index pointer over
+            time.sleep(self.speed(.5))
+            for item in indexDisplay:
+                self.canvas.move(item, CELL_SIZE, 0)
+            self.window.update()
+            # wait a moment so the user can see the index pointer moved
+            time.sleep(self.speed(.5))

```

Comment:
These 5 lines of code should be replaced with a call to the `moveItemsBy()` method.  That will make the movement a bit smoother and handle user interaction via calls to `wait()`
`self.moveItemsBy(indexDisplay, (CELL_SIZE, 0), sleepTime=0.03)`
I also think the sleepTime of 0.03 is more reasonable for fast animation.